perfkitbenchmarker/traces/dstat.py

```diff
@@ -112,10 +123,49 @@ def Start(self, sender, benchmark_spec):
                                      str(uuid.uuid4())[:8])
     start_on_vm = functools.partial(self._StartOnVm, suffix=suffix)
     vm_util.RunThreaded(start_on_vm, benchmark_spec.vms)
+    self._start_time = time.time()
 
   def Stop(self, sender, benchmark_spec):
     """Stop dstat on all VMs in 'benchmark_spec', fetch results."""
-    vm_util.RunThreaded(self._StopOnVm, benchmark_spec.vms)
+    events.record_event.send(sender, event='dstat',
+                             start_timestamp=self._start_time,
+                             end_timestamp=time.time(),
+                             metadata={})
+    args = []
+    for role, vms in benchmark_spec.vm_groups.iteritems():
+      args.extend([((
+          vm, '%s_%s' % (role, idx)), {}) for idx, vm in enumerate(vms)])
+    vm_util.RunThreaded(self._StopOnVm, args)
+
+  def Analyze(self, sender, samples):
+    """Analyze dstat file and record samples."""
+
+    def _AnalyzeEvent(role, labels, out, event):
+      for idx, label in enumerate(labels[1:], 1):
+        cond = [(out[:, 0] > event.start_timestamp) & (
+            out[:, 0] < event.end_timestamp)]
+        choice = [out[:, idx]]
+        filtered = np.select(cond, choice)
+        avg = np.mean(filtered[np.nonzero(filtered)])
```

Comment:
Why do the computation on one column at a time instead of all of them at the same time?